lib/dao.js

```diff
@@ -1413,11 +1413,17 @@ DataAccessObject.removeById = DataAccessObject.destroyById = DataAccessObject.de
 
   var Model = this;
 
-  this.remove(byIdQuery(this, id).where, options, function(err) {
+  this.remove(byIdQuery(this, id).where, options, function(err, data) {
+    var deleted = data && data.count > 0;
     if ('function' === typeof cb) {
-      cb(err);
+      if (options.reportFailureAsError && !err && !deleted) {
+        err = new Error('No instance with id ' + id + ' found for ' + Model.modelName);
+        cb(err, false);
+      } else {
+        cb(err, deleted);
```

Comment:
That's the thing, you cannot project the future and know what we will need later. A made-up example: let's say we add support for cascading deletes that is not implemented by all connectors and you want to signal back whether the cascading delete was performed or not. You can get `{ count: 1, cascaded: true }`, `{count: 1, cascaded: false }`, but also `{ count: 0 }` or `{ count: undefined }`. 

> I don't want to end up doing: data && data.success

Then put this check into dao.js, so that the cb arg is always set.

```
cb(err, { count: data ? data.count : undefined })
```